shared/js/background/features/fire-button.js

```diff
@@ -0,0 +1,305 @@
+            selectedOption: this.settings.getSetting('fireButtonDefaultOption') || 'CurrentSite',
+            since: undefined
+        }
+    }
+
+    /**
+     * Burn browser storage as per the config options passed to the function.
+     * @param {Partial<BurnConfig>} options
+     * @returns {Promise<boolean>}
+     */
+    async burn (options) {
+        /** @type {BurnConfig} config */
+        const config = Object.assign(this.getDefaultSettings(), options)
+
+        console.log('🔥', config)
+        try {
+            await this.clearTabs(config.closeTabs, config.origins)
+            // 1/ Clear downloads and history
+            const clearing = []
+            if (!config.origins || config.origins.length === 0) {
+                clearing.push(chrome.browsingData.remove({
+                    since: config.since
+                }, {
+                    downloads: true,
+                    history: config.clearHistory
+                }))
+            }
+            // TODO: handle clearing downloads and history for specific origins
+
+            // 2/ Clear cookies, except on SERP
+            const cookieOptions = {
+                since: config.since
+            }
+            if (config.origins) {
+                cookieOptions.origins = config.origins
+            } else {
+                cookieOptions.excludeOrigins = ['https://duckduckgo.com']
+            }
+            clearing.push(chrome.browsingData.remove(cookieOptions, {
+                cookies: true,
+                localStorage: true
+            }))
+            // 3/ Clear origin-keyed storage
+            clearing.push(chrome.browsingData.remove({
+                origins: config.origins,
+                since: config.since
+            }, {
+                appcache: true,
+                cache: true,
+                cacheStorage: true,
+                indexedDB: true,
+                fileSystems: true,
+                serviceWorkers: true,
+                webSQL: true
+            }))
+            const results = await Promise.all(clearing)
+            console.log('🔥 result', results)
+            sendPixelRequest('e_firebutton_click', {
+                atb: this.settings.getSetting('atb')
+            })
+            return true
+        } catch (e) {
+            console.error('🔥 error', e)
+            return false
+        }
+    }
+
+    /**
+     * @param {boolean} closeTabs If true, tabs are also closed
+     * @param {string[]} [origins] Only close tabs for these origins
+     */
+    async clearTabs (closeTabs = true, origins) {
+        // gather all non-pinned tabs
+        const openTabs = (await browser.tabs.query({
+            pinned: false
+        })).filter(tabMatchesHostFilter(origins))
+        const removeTabIds = openTabs.map(t => t.id || 0)
+        // clear adclick attribution data
+        removeTabIds.forEach((tabId) => {
+            const tab = this.tabManager.tabContainer[tabId]
```

Comment:
Nit:
```suggestion
            this.tabManager.tabContainer[tabId]?.adClick = null
```

Reply:
You can't use optional chaining on the left-hand side of an expression in JS: `SyntaxError: Invalid left-hand side in assignment`